cmd/kubeadm/app/phases/upgrade/configuration.go

```diff
@@ -103,9 +102,14 @@ func bytesToValidatedMasterConfig(b []byte) (*kubeadmapiext.MasterConfiguration,
 	finalCfg := &kubeadmapiext.MasterConfiguration{}
 	internalcfg := &kubeadmapi.MasterConfiguration{}
 
-	if err := runtime.DecodeInto(legacyscheme.Codecs.UniversalDecoder(), b, cfg); err != nil {
+	decoded, err := kubeadmapiext.LoadYAML(b)
```

Comment:
What about implementing `kubeadmapiext.FromBytes()`?
I kind of think that the loading of the generic structure and the migration into the target format should be hidden into `kubeadm/app/apis/kubeadm/v1alpha1` package....